packages/template-retail-react-app/app/pages/product-detail/index.jsx

```diff
@@ -155,10 +163,31 @@ const ProductDetail = ({category, product, isLoading}) => {
                         <ProductView
                             product={product}
                             category={primaryCategory?.parentCategoryTree || []}
-                            addToCart={(variant, quantity) => handleAddToCart(variant, quantity)}
+                            canOrder={setProductsSelected}
+                            addToCart={() => {
+                                const productSelectionValues = Object.values(productSetSelection)
+
+                                return handleAddToCart(productSelectionValues)
+                            }}
                             addToWishlist={(_, quantity) => handleAddToWishlist(quantity)}
                             isProductLoading={isLoading}
                             isCustomerProductListLoading={!wishlist.isInitialized}
+                            validate={() => {
```

Comment:
I've removed the `canOrder` prop in favour of using a new property called `validate`. This prop which is a function allows you to add custom logic to whether or not a product should run it's handleAddToCart. We needed this for the parent product set product because you can only add it to the cart if all the selections of it's child are made, and that logic can't be placed in the ProductView component, it must be in the page itself. Return true or false depending on the validity, I maintain the default canOrder logic as a default prop value.

Reply:
I'm not sure if `validate` is clear on its own _what_ is being validated. Maybe call it `validateOrder` or something?

Reply:
Or maybe `shouldAddToCart`..

Reply:
I changed this to `validateAttributeSelection`, a little verbose, but it works I think.